### PR TITLE
Babel Preset Default: Configure @babel/preset-env preset to respect a local Browserslist configuration

### LINKED ISSUE #31857: Babel Preset Default: Allow to use a custom browserslist config
## What problem does this address?
When using a custom browserslist config with `@wordpress/scripts` it's currently only used for (Post)CSS. For JavaScript the `@wordpress/browserslist-config` config will still be used because it's defined in `@wordpress/babel-preset-default`:

https://github.com/WordPress/gutenberg/blob/08faf92b37f1dadb772b5399a77b310444e0cae7/packages/babel-preset-default/index.js#L29-L31

This is a bit unexpected.

## What is your proposed solution?

**First idea**: Allow to extend the options passed to `@babel/preset-env`. Something like this:

```json
  "babel": {
    "presets": [
      [ "@wordpress/babel-preset-default", { "presetEnvOpts": { "targets": "", "debug": true } } ]
    ],
  }
```
`getPresetEnv()` would merge the custom `presetEnvOpts` when passing them to `@babel/preset-env`.

**Second idea**: Check if a browserslist config is available. Something like [`hasBabelConfig()`](https://github.com/WordPress/gutenberg/blob/a542c0deebfb439c2667ff372aa1e16a5f001cbc/packages/scripts/utils/config.js#L18-L25) in `@wordpress/scripts`. If set, leave `targets` empty so  `@babel/preset-env` can pick up the project's config. Otherwise use `@wordpress/browserslist-config` as right now.

**Third idea**: More of a workaround. Keep it as is and require devs to add `@babel/preset-env` as a second preset which also seems to override the config from `@wordpress/babel-preset-default`:

```json
  "babel": {
    "presets": [
      "@wordpress/babel-preset-default"
      [ "@babel/preset-env", { "modules": false, "debug": true } ]
    ],
  }
```

Any thoughts and/or other ideas?

### PR DESCRIPTION
**Pull request by ocean90**
## Description
This uses [browserslist.findConfig()](https://github.com/browserslist/browserslist/blob/209adf9e0051fa39a2b25354cffd493300f34b02/node.js#L301) to first search for a local config before falling back to `@wordpress/browserslist-config`. 

Fixes #31857.

## How has this been tested?
To verify what targets are used you can set `opts.debug = true`  after line 37

Before:
```
Using polyfills: No polyfills were added, since the `useBuiltIns` option was not set.
@babel/preset-env: `DEBUG` option

Using targets:
{
  "android": "90",
  "chrome": "89",
  "edge": "89",
  "firefox": "87",
  "ios": "14",
  "opera": "74",
  "safari": "14",
  "samsung": "13"
}
```

After (with `@wearerequired/browserslist-config`):
```
Using polyfills: No polyfills were added, since the `useBuiltIns` option was not set.
@babel/preset-env: `DEBUG` option

Using targets:
{
  "chrome": "89",
  "edge": "89",
  "firefox": "87",
  "ie": "11",
  "ios": "14",
  "opera": "74",
  "safari": "14"
}
```

## Types of changes
Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -13160,7 +13160,41 @@
 				"@wordpress/browserslist-config": "file:packages/browserslist-config",
 				"@wordpress/element": "file:packages/element",
 				"@wordpress/warning": "file:packages/warning",
+				"browserslist": "^4.16.6",
 				"core-js": "^3.6.4"
+			},
+			"dependencies": {
+				"browserslist": {
```
Comment (gziolo) [inline package-lock.json]: Can we bump `browserslist` version in other places? Here:
https://github.com/WordPress/gutenberg/blob/b5f6955742e784f11828e1dacba4361713135963/package.json#L154

**Review by gziolo — COMMENTED** [review]
I haven't tested it yet but it looks good. I hope we merge it today 👍🏻

File: package-lock.json (hunk 1 above)
Comment (ocean90) [reply package-lock.json]: Updated in https://github.com/WordPress/gutenberg/pull/31898/commits/9898955bf5e2bc8995970da736983f05f9966920 with these steps:

1. `npm install browserslist@latest --save-dev`
2. Removed browserslist from the package again
3. `npm i`
4. `lerna add browserslist packages/babel-preset-default`

Not sure if steps 2-4 were necessary, probably not.

Comment (ocean90) [reply package-lock.json]: I guess this section should have been removed with the update though? Is it okay to remove it manually or did I do something wrong?

Comment (gziolo) [reply package-lock.json]: Yes, updating mannually `dependencies` key and its value should do the trick. You only need to run `npm i` afterward to ensure it's all good.

Comment (ocean90) [reply package-lock.json]: Thanks, looks good now. I also updated `caniuse-lite` with `npx browserslist@latest --update-db` which did remove the extra `caniuse-lite` dependency.

```
❯ npx browserslist@latest --update-db
npx: Installierte 6 in 1.414s
Latest version:     1.0.30001228
Installed versions: 1.0.30001214, 1.0.30001228
Removing old caniuse-lite from lock file
....

❯ npx browserslist@latest --update-db
npx: Installierte 6 in 1.316s
Latest version:     1.0.30001228
Installed version:  1.0.30001228
caniuse-lite is up to date
```

**Review by gziolo — APPROVED** [review]
Cool, let's get this in to have it included in Gutenberg 5.7. I can test later.

**Comment by gziolo** [comment]
In fact, it's going to be useful for Gutenberg itself because it will finally use the top-level config :)

https://github.com/WordPress/gutenberg/blob/trunk/.browserslistrc

I tested locally and it loads it correctly now 👍🏻

**Pull request merged**